frontend/src/tests/instructor-permission-test.js

```diff
@@ -53,54 +206,238 @@ export function instructorsPermissionTests(api) {
         expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET(`/instructor/sessions/${session.id}/positions`);
+        expect(resp).toHaveStatus("success");
     });
 
-    it.todo("can't update position");
+    it("can't update position", async () => {
+        await switchToInstructorOnlyUser();
+
+        const newPosition = {
+            position_code:
+                "instructor_permission_cant_update_position_test_code",
+            position_title:
+                "instructor permission cant update position test position",
+            hours_per_assignment: 1.0,
+            contract_template_id: existingContractTemplateId,
+            duties: "Tutorials",
+        };
+
+        let resp = await apiPOST(
+            `/instructor/sessions/${session.id}/positions`,
+            newPosition
+        );
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/positions`,
+            newPosition
+        );
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(`/admin/positions`, newPosition);
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(`/instructor/positions`, newPosition);
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
 
     it("fetch contract templates", async () => {
+        await switchToInstructorOnlyUser();
         let resp = await apiGET(
             `/instructor/sessions/${session.id}/contract_templates`
         );
         expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET(
+            `/instructor/sessions/${session.id}/contract_templates`
+        );
+        expect(resp).toHaveStatus("success");
     });
 
-    it.todo("can't update contract template");
+    it("can't update contract template", async () => {
+        await switchToInstructorOnlyUser();
+
+        const newTemplate = {
+            template_file: "default-template.html",
+            template_name: "Test",
+        };
+
+        let resp = await apiPOST(
+            `/instructor/sessions/${session.id}/contract_templates`,
+            newTemplate
+        );
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/contract_templates`,
+            newTemplate
+        );
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
 
     it("fetch applicants", async () => {
+        await switchToInstructorOnlyUser();
         let resp = await apiGET(
             `/instructor/sessions/${session.id}/applicants`
         );
         expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+        resp = await apiGET(`/instructor/sessions/${session.id}/applicants`);
+        expect(resp).toHaveStatus("success");
```

Comment:
Are you going to test that the correct applicants show up?

Reply:
I would argue checking the content of the response should be checked inside the `instructor-tests.js`, while `intructor-permission-tests.js` only checks the accessibility of different routes given a certain set of permissions.